### DISCUSSION TITLE
NRF52-DK support

### DISCUSSION THREAD

**Discussion by SL06** [Bodmer/TFT_eSPI#2211] [New processor/display support or capability request]
HI,  I saw taht the library suppport generic processor andwas hopping that the Nordic nrf52-DK  + ILI9486 8bit parallel display might work.  The nrf52-DK is same format as the arduino Uno and I can insert the display directly.  When I try to compile exemples, I got the following errors. Any suggestion ?   

/Users/sl/Documents/Arduino/libraries/TFT_eSPI/TFT_eSPI.cpp: In member function 'void TFT_eSPI::initBus()':
/Users/slDocuments/Arduino/libraries/TFT_eSPI/TFT_eSPI.cpp:584:3: error: 'PARALLEL_INIT_TFT_DATA_BUS' was not declared in this scope
   PARALLEL_INIT_TFT_DATA_BUS;
   ^
/Users/sl/Documents/Arduino/libraries/TFT_eSPI/TFT_eSPI.cpp: In member function 'uint8_t TFT_eSPI::readcommand8(uint8_t, uint8_t)':
/Users/sl/Documents/Arduino/libraries/TFT_eSPI/TFT_eSPI.cpp:993:10: error: 'GPIO_DIR_MASK' was not declared in this scope
   busDir(GPIO_DIR_MASK, INPUT);
          ^
/Users/sl/Documents/Arduino/libraries/TFT_eSPI/TFT_eSPI.cpp: In member function 'virtual uint16_t TFT_eSPI::readPixel(int32_t, int32_t)':
/Users/s/Documents/Arduino/libraries/TFT_eSPI/TFT_eSPI.cpp:1078:10: error: 'GPIO_DIR_MASK' was not declared in this scope
   busDir(GPIO_DIR_MASK, INPUT);
          ^
/Users/sl/Documents/Arduino/libraries/TFT_eSPI/TFT_eSPI.cpp: In member function 'void TFT_eSPI::readRect(int32_t, int32_t, int32_t, int32_t, uint16_t*)':
/Users/sl/Documents/Arduino/libraries/TFT_eSPI/TFT_eSPI.cpp:1204:10: error: 'GPIO_DIR_MASK' was not declared in this scope
   busDir(GPIO_DIR_MASK, INPUT);

**Comment by Bodmer** [comment]
Only the SPI interface to a TFT is supported for generic processors.

**Comment by SL06** [comment]
Thanks for the answer.

Sylvain

**Comment by SL06** [comment]
Hi,

I have written the processor section for my NRF52 DK board - 8 bit parallel with ILI9486 TFT.   Its working but not perfectly - see picture.
The region 0, 0, 240, 240p (approx)  seems OK but text or picture (jpeg) exceeding that region shows some artefact.  

FillScreen does not seem affected.  Screen rotation does not have an effect on the artefact.

I have played with write8 and write16 timing by adding delay here and there without improvement.  

Suggestions as to where to look would be appreciated.

Happy holidays,


 























Sylvain Lareau  (Le Doc du Vélo)
Ledocduvelo.ca <http://ledocduvelo.ca/>
***@***.*** ***@***.***>
(819) 307-5117
